controllers/machine_controller.go

```diff
@@ -249,32 +248,33 @@ func (r *MachineReconciler) getMachinesInCluster(ctx context.Context, namespace,
 
 // isDeleteReady returns an error if any of Boostrap.ConfigRef or InfrastructureRef referenced objects still exist.
 func (r *MachineReconciler) isDeleteReady(ctx context.Context, m *clusterv1.Machine) error {
-	if m.Spec.Bootstrap.ConfigRef != nil {
-		_, err := external.Get(r.Client, m.Spec.Bootstrap.ConfigRef, m.Namespace)
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
-		if err != nil {
-			return errors.Wrapf(err, "failed to get %s %q for Machine %q in namespace %q",
-				path.Join(m.Spec.Bootstrap.ConfigRef.APIVersion, m.Spec.Bootstrap.ConfigRef.Kind),
-				m.Spec.Bootstrap.ConfigRef.Name, m.Name, m.Namespace)
-		}
-		return errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: 10 * time.Second},
-			"delete is not ready, Bootstrap configuration still exists")
-	}
+	bootstrapConfigReady := r.isExternalDeleteReady(ctx, m.Namespace, m.Spec.Bootstrap.ConfigRef)
+	infraReady := r.isExternalDeleteReady(ctx, m.Namespace, &m.Spec.InfrastructureRef)
 
-	if _, err := external.Get(r.Client, &m.Spec.InfrastructureRef, m.Namespace); err != nil && !apierrors.IsNotFound(err) {
-		return errors.Wrapf(err, "failed to get %s %q for Machine %q in namespace %q",
-			path.Join(m.Spec.InfrastructureRef.APIVersion, m.Spec.InfrastructureRef.Kind),
-			m.Spec.InfrastructureRef.Name, m.Name, m.Namespace)
-	} else if err == nil {
+	if !bootstrapConfigReady || !infraReady {
 		return errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: 10 * time.Second},
-			"delete is not ready, Infrastructure configuration still exists")
+			"delete is not ready, Bootstrap configuration or Infrastructure configuration still exists")
 	}
 
 	return nil
 }
 
+func (r *MachineReconciler) isExternalDeleteReady(ctx context.Context, namespace string, ref *corev1.ObjectReference) bool {
```

Comment:
@ncdc how do you feel about having one function for both refs like this rather than `isBootstrapConfigDeleteReady`, etc?